### ISSUE TITLE
AuthUserAttribute always attempts to convert String to Int

### ISSUE THREAD

**Issue by look1n** [aws-amplify/amplify-flutter#1164]
**Describe the bug**
Custom user attributes forced conversion to int

**To Reproduce**
1. Create custom user attribute with String type
2. Save custom user attribute with string value that contains valid number i.e "012345"
3. Fetch user attributes
4. See that created custom attribute value has int type and value is 12345

**Expected behavior**
Custom attribute value should have same type specified on custom attribute creation form.

**Platform**
[] Android
[] iOS

**Additional context**
Obvious reason of such behavior is AuthUserAttribute init function that forcefully cast any string value that holds a number into int
 ```
AuthUserAttribute.init({required this.userAttributeKey, required this.value}) {
    this.userAttributeKey = userAttributeKey;
    if (userAttributeKey != 'phone_number') {
      try {
        this.value = int.parse(value);
      } on FormatException {}
    }
}
```

**Comment by Jordan-Nelson** [comment]
Hello @look1n - Thanks for opening the issue. 

Some context - I believe the underlying API that is being used sends all attributes back as strings, without the information about the type specified in Cognito. The code you referenced above is (incorrectly) attempting to convert the attribute to the correct type.

The int conversion code leads to unpredictable behavior, so I think this is clearly a bug and can be removed. Unfortunately this will result in all attributes being returned as Strings, but at least the behavior will be predictable so I think this is an improvement. This would be a breaking change since some applications may be relying on that automatic conversion. I can take this back and see when we would be able to implement this.

As you suggested in the expected behavior, the better fix would be to convert the attribute to the type that was specified in Cognito. I think this will require some more investigation on our part to see how we could obtain the type info from Cognito.

**Comment by Jordan-Nelson** [comment]
@look1n - The change to remove `int.parse()` was included in the latest release candidate for the next major release, `0.3.0-rc.3`. It will be included in the first stable version of `0.3.0` when it is released.

**Comment by Jordan-Nelson** [comment]
@look1n - v0.3.0 has been released, which includes this fix. Please note that v0.3.0 does include several breaking changes (this being one of them). Please refer to the change log for details - https://pub.dev/packages/amplify_flutter/changelog.

As mentioned above, all attributes will now be returned as a String, regardless of the type defined in Cognito (String or Bool). I opened a [separate issue](https://github.com/aws-amplify/amplify-flutter/issues/1288) to track the feature request to have the types match the definition in Cognito console. I am going to close this issue out.
